src/webgpu/shader/execution/robust_access_vertex.spec.ts

```diff
@@ -0,0 +1,40 @@
+export const description = `
+Test vertex attributes behave correctly (no crash / data leak) when accessed out of bounds
+`;
+
+/*
+Test coverage:
+
+The following will be parameterized (all combinations tested):
+
+1) Draw call indexed? (false / true)
+  - Run the draw call using an index buffer
+
+2) Draw call indirect? (false / true)
+  - Run the draw call using an indirect buffer
+
+3) Draw call parameter (vertexCount, firstVertex, indexCount, firstIndex, baseVertex, instanceCount,
+  firstInstance)
+  - The parameter which will go out of bounds. Filtered depending on if the draw call is indexed.
+
+4) Attribute type (float, vec2, vec3, vec4, mat2, mat3, mat4)
+  - The input attribute type in the vertex shader
+
+5) Error scale (1, 100, 100000000)
+  - Offset to add to the correct draw call parameter
+
+6) Additional vertex buffer (false / true)
+  - Tests that no OOB occurs if another vertex buffer is bound
+
+The tests will also have another vertex buffer bound for an instanced attribute, to make sure
+instanceCount / firstInstance are tested.
+
+The test will run a render pipeline which verifies the following:
+1) All vertex attribute values are in-bounds (equal to zero, since buffers will be zero'd)
```

Comment:
Can the test check that values sampled in bounds have the correct value (non zero, they should be pretty specific values, maybe randomly generated? @kainino0x maybe we should have repeatable random generators?), and that the value sampled OOB has one of the allowed values?

Reply:
We probably should have some simple repeatable random generators, @iraiter2 feel free to add some in a common place (probably a new file in framework/)

Reply:
Can't values be equal to zero, or some other value inside the buffer?

Reply:
I'm thinking if the values can be either zero or a value in the buffer, filling the buffer with random info doesn't detect anything more? With a buffer full of zeros, if the value we see is anything non-zero, we can be sure it didn't come from inside the buffer. Is there a case where we would detect an invalid RBA implementation if we filled the buffer with data compared to if it was full of zeros?

Reply:
Ah, I forgot about the exact RBA rules and that it allows things like % the index. I still think it's useful to have non-zero data in the vertex buffer (maybe just 1, 2, 3, 4, 5, ...) for two reasons:

 - We can check that in-bounds access return the correct value, which means the test doesn't always sees zeroes because of a bug, and passes because of that.
 - By looking at the values, we can learn about how RBA is implemented on different hardware. (and maybe constrain it to be only clamping in the future?)

Reply:
Sounds good. Maybe it would be best to have just a few random values throughout, so that a real OOB read isn't accidentally marked valid?

Reply:
Talking about it we think a few values repeated for now would work, and randomness can be added if we think conflicting buffers could exist from previous tests. But, the random values should be repeatable so that may be more complicated.